### ISSUE TITLE
[v4 a2] Regression of <DeleteButton /> as a child of <DataGrid />

### ISSUE THREAD

**Issue by smeng9** [marmelab/react-admin#7190]
**What you were expecting:**
DeleteButton should be rendered and functional
![Screenshot from 2022-02-04 16-01-27](https://user-images.githubusercontent.com/38666763/152609138-bbf30e6e-7edd-44f0-81fa-8b3364a6cb56.png)

**What happened instead:**
DeleteButton does not get rendered
![Screenshot from 2022-02-04 16-00-11](https://user-images.githubusercontent.com/38666763/152609001-1be6561b-135c-474e-8f2d-63635f751d18.png)

**Steps to reproduce:**
1. Fix the react-hook-form dependency issue in examples/tutorial in next branch
2. Add the `<DeleteButton />` to children of the `<Datagrid />` component in `posts.js`

**Related code:**

```
https://codesandbox.io/s/romantic-proskuriakova-h60n2?file=/src/posts.js:920-932
```

**Other information:**

**Environment**

* React-admin version: 4.0.0-alpha.2
* Last version that did not exhibit the issue (if applicable): 3.19.7
* React version: 17.0.2
* Browser: Firefox 96
* Stack trace (in case of a JS error): No error

**Comment by fzaninotto** [comment]
Reproduced, thanks.

**Comment by fzaninotto** [comment]
Fixed by #7198
